packages/vm/src/runTx.ts

```diff
@@ -619,6 +671,20 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
     }
   }
 
+  /**
+   * Cleanup code of accounts written to in a 7702 transaction
+   */
+
+  for (const str of writtenAddresses) {
+    const address = Address.fromString(str)
+    // Special case: in StateManager if we try to set code to the empty code, we ignore this
+    // Do not do this and explicitly override
+    // TODO maybe use modifyAccount?
+    const account = (await this.stateManager.getAccount(address)) ?? new Account()
+    account.codeHash = KECCAK256_NULL
+    await this.stateManager.putAccount(address, account)
```

Comment:
This should work just fine (`modifyAccountFields` already generates a new account if it doesn't exist in the state).
```suggestion
    await this.stateManager.modifyAccountFields(address, { codeHash: KECCAK256_NULL })
```

Reply:
I don't understand why I did not do this directly :thinking:

Reply:
Ah the original reason was that I wanted to `putContractCode` to empty, which is thus ignored. But I just added a test and it still fails :thinking: